pkgs/applications/networking/cluster/tektoncd-cli/default.nix

```diff
@@ -13,35 +13,39 @@ buildGoModule rec {
 
   vendorSha256 = null;
 
-  doCheck = false;
+  subPackages = [ "cmd/tkn" ];
+
+  buildFlagsArray = [
+    "-ldflags="
+    "-s"
+    "-w"
+    "-X github.com/tektoncd/cli/pkg/cmd/version.clientVersion=${version}"
+  ];
 
   nativeBuildInputs = [ installShellFiles ];
 
-  buildPhase = ''
-    make bin/tkn
+  checkPhase = ''
+    make test-unit
```

Comment:
🎉

Reply:
Although, I think relying on `go test` (that `buildGoModule` does automatically right ?) should work fine.

Reply:
Leaving `checkPhase` to `buildGoModule` said no tests found as it only looks in `cmd/tkn`

```
running tests
?       github.com/tektoncd/cli/cmd/tkn [no test files]
```

I'll try some different `subPackages` options

Reply:
If you don't set `subPackages`:

```
Building subPackage ./third_party/VENDOR-LICENSE/github.com/hashicorp/errwrap
main module (github.com/tektoncd/cli) does not contain package github.com/tektoncd/cli/third_party/VENDOR-LICENSE/github.com/hashicorp/errwrap
error: --- Error
```

Reply:
`rm -rf third_party` lets it run pretty well. Hitting a couple failing tests with the version

Should I sed `Client version: dev` -> `Client version: 0.16.0` ?

Alternatively I could update the golden files then testing it matches the golden files, but that seems useless

```
=== RUN   TestGetVersions/empty_deployment_items
    version_test.go:231: assertion failed:
        --- expected
        +++ actual
        @@ -1,3 +1,3 @@
        -Client version: dev
        +Client version: 0.16.0
         Pipeline version: unknown, pipeline controller may be installed in another namespace please use tkn version -n {namespace}



        You can run 'go test . -test.update-golden' to automatically update testdata/TestGetVersions-empty_deployment_items.golden to the new expected value.'

=== RUN   TestGetVersions/deployment_only_with_pipeline_installed
    version_test.go:231: assertion failed:
        --- expected
        +++ actual
        @@ -1,3 +1,3 @@
        -Client version: dev
        +Client version: 0.16.0
         Pipeline version: v0.10.0



        You can run 'go test . -test.update-golden' to automatically update testdata/TestGetVersions-deployment_only_with_pipeline_installed.golden to the new expected value.'

=== RUN   TestGetVersions/deployment_with_pipeline_and_triggers_installed
    version_test.go:231: assertion failed:
        --- expected
        +++ actual
        @@ -1,3 +1,3 @@
        -Client version: dev
        +Client version: 0.16.0
         Pipeline version: v0.10.0
         Triggers version: v0.5.0


        You can run 'go test . -test.update-golden' to automatically update testdata/TestGetVersions-deployment_with_pipeline_and_triggers_installed.golden to the new expected value.'

=== RUN   TestGetVersions/deployment_with_pipeline_and_dashboard_installed
    version_test.go:231: assertion failed:
        --- expected
        +++ actual
        @@ -1,3 +1,3 @@
        -Client version: dev
        +Client version: 0.16.0
         Pipeline version: v0.10.0
         Dashboard version: v0.7.0


        You can run 'go test . -test.update-golden' to automatically update testdata/TestGetVersions-deployment_with_pipeline_and_dashboard_installed.golden to the new expected value.'

=== RUN   TestGetVersions/deployment_with_pipeline,_triggers_and_dashboard_installed
    version_test.go:231: assertion failed:
        --- expected
        +++ actual
        @@ -1,3 +1,3 @@
        -Client version: dev
        +Client version: 0.16.0
         Pipeline version: v0.10.0
         Triggers version: v0.5.0


        You can run 'go test . -test.update-golden' to automatically update testdata/TestGetVersions-deployment_with_pipeline,_triggers_and_dashboard_installed.golden to the new expected value.'

--- FAIL: TestGetVersions (0.00s)
    --- FAIL: TestGetVersions/empty_deployment_items (0.00s)
    --- FAIL: TestGetVersions/deployment_only_with_pipeline_installed (0.00s)
    --- FAIL: TestGetVersions/deployment_with_pipeline_and_triggers_installed (0.00s)
    --- FAIL: TestGetVersions/deployment_with_pipeline_and_dashboard_installed (0.00s)
    --- FAIL: TestGetVersions/deployment_with_pipeline,_triggers_and_dashboard_installed (0.00s)
```

Reply:
That worked, all tests passed. I'll commit and push. LMK if there's a cleaner way to do this :)

Reply:
Would it be worth `rm -rf` all the .go files in VENDOR-LICENSE in https://github.com/tektoncd/cli?
Should I post a bug report in https://github.com/google/go-license?
Or is including the .go files a requirement of the license? Looks like it's all just MPL-2.0 License though.

```
λ find ./third_party/VENDOR-LICENSE -type f -name "*.go"
./third_party/VENDOR-LICENSE/github.com/hashicorp/errwrap/errwrap.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/errwrap/errwrap_test.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/golang-lru/2q_test.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/golang-lru/2q.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/golang-lru/simplelru/lru.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/golang-lru/simplelru/lru_test.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/golang-lru/simplelru/lru_interface.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/golang-lru/lru.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/golang-lru/arc.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/golang-lru/doc.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/golang-lru/lru_test.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/golang-lru/arc_test.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-multierror/multierror_test.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-multierror/sort_test.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-multierror/prefix.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-multierror/flatten_test.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-multierror/append.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-multierror/group.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-multierror/multierror.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-multierror/format_test.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-multierror/sort.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-multierror/prefix_test.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-multierror/flatten.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-multierror/append_test.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-multierror/format.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-multierror/group_test.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-version/version_collection_test.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-version/constraint.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-version/version_collection.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-version/constraint_test.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-version/version.go
./third_party/VENDOR-LICENSE/github.com/hashicorp/go-version/version_test.go
```

Reply:
oh I didn't think of that 😅 (both for the subpackage and the tests that check version)

Reply:
I think it looks good 😉

Reply:
:+1: If these files are removed from https://github.com/tektoncd/cli at some point we can drop the line

Reply:
@vdemeester Not sure where else to ask but is it possible to change the release note of a merged PR before release?